### ISSUE TITLE
[Blazor ASP.NET Core hosted] Conditionally launch client side Blazor app.

### ISSUE THREAD

**Issue by rxelizondo** [aspnet/AspNetCore#10631]
Feature request for `Blazor (ASP.NET Core hosted)` solutions (consisting of a client side Blazor app project plus server side / WebAPI project).

I have a domain address such as `contoso.com`. When people visit `contoso.com` I would like to detect if the visitor has been authenticated to use my Blazor app and then:

1) If the visitor has been authenticated, proceed and automatically launch the client side Blazor app.

2) If the visitor has not  been authenticated, I would like the `contoso.com` endpoint to show a simple webpage with product information where visitors can learn about my app and a big button that the visitor can click to launch the client side Blazor app. 

The bottom line is that I really don’t want to launch the client side Blazor app when someone is just visiting the `contoso.com` address because the app may take a very long time to load and startup specially in slow phones with bad connections. It is only until the visitor has **explicitly** chosen to launch the app or has **already been authenticated** to the app that I wish the client side Blazor app to be launched.

I suspect this will be a common scenario for client side Blazor apps, so it would be nice if the functionality was built-in as part of the Blazor infrastructure.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @rxelizondo.
You can do this with routing - and designate different endpoints for your authenticated and non-authenticated pages. And sure one of those can be Blazor, and the other - not.

**Comment by rxelizondo** [comment]
@mkArtakMSFT - Thank you very much for your help, however, I am not sure how you can achieve such functionality with the routing, but maybe I am missing something, walking through the scenario, it would work something like this:

1) Visitor enters `contoso.com` in the browser.
2) The request is sent to the server and ASP.NET begins processing it.
3) Because there is no `Controller/Action` specified in `contoso.com`, the default will be `Home/Index`.
4) Assuming that I have implemented the `Home/Index` action, it will then be called by the routing engine.
5) Once `Home/Index` is executing, I can detect if the  user is authenticated or not, if not, I can redirect the request to some informational page, but what happens if the user is authenticated? then what? How do I automatically start the Blazor client side app?

Is that what you had in mind or did I totally missed the point?

Thanks.
